AK/URLParser.cpp

```diff
@@ -1323,7 +1323,7 @@ URL URLParser::basic_parse(StringView raw_input, Optional<URL> const& base_url,
                     // 2. If the code point substring from pointer to the end of input does not start with a Windows drive letter, then shorten url’s path.
                     auto substring_from_pointer = input.substring_view(iterator - input.begin()).as_string();
                     if (!starts_with_windows_drive_letter(substring_from_pointer)) {
-                        if (!url->m_paths.is_empty() && !(url->scheme() == "file" && url->m_paths.size() == 1 && is_normalized_windows_drive_letter(url->m_paths[0])))
+                        if (!url->m_paths.is_empty())
```

Comment:
I think this may have to do with the [shorten a url's path](https://url.spec.whatwg.org/#shorten-a-urls-path) steps. We could do away with this if check and instead move these steps into its own function since it is confusing to see this logic without any explicit mention of it in this step. The steps are also used elsewhere in the parser.

Reply:
thanks! Yeah, I think we have this mistake repeated elsewhere